src/mscorlib/shared/System/Memory.cs

```diff
@@ -310,7 +310,7 @@ public bool Equals(Memory<T> other)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override int GetHashCode()
         {
-            return CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
+            return CombineHashCodes(_object == null ? 0 : _object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
```

Comment:
Do we care about the case where _object == null but _index or _length are not 0?  Seems like that should only happen in the case of mischief or something going terribly wrong somewhere, in which case this could instead be:
```C#
public override int GetHashCode() => _object != null ?
    CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode()) :
    0;
```